### DISCUSSION TITLE
Backend not connecting to frontend

### DISCUSSION THREAD

**Discussion by aircliu** [vercel/community#3659] [Help] [answered]
### Summary

Hi everyone help is needed!, 

I am trying to deploy my website and for context I have a project with my backend connected, and a project for my front end. Right now after i deploy both ends, the backend is not communicating with the front end, so for example, when we are trying to login to our accounts, when we press login the backend doesn't receive anything and we are just stuck there.

 I inspected the console and it looks like there is a CORS error, but I doubled checked with my partner to make sure it is right. Also I made sure the env variables were correctly pasted into vercel.
 
<img width="397" alt="image" src="https://github.com/vercel/community/assets/109885692/64be4f00-d011-40de-a87b-329b7a168f7e">

Any tips is appreciated!

### Example

https://geniusai.education/login

### Steps to Reproduce

1. when you go onto the page, try to register an account, after you press register it should redirect you to https://geniusai.education/landing

code so far:

app = Flask(__name__)
app.secret_key = os.getenv('SECRETKEY')
openai.organization = os.environ['OPENAI_ORG']
openai.api_key = os.environ['OPENAI_API_KEY']
# Tell 3001 to expect requests from 3000
CORS(app, resources={r"/api/*": {"origins": os.getenv('REACT_APP_PROD_HOST')}}, supports_credentials=True)

cursor = db.cursor()

// made sure react_app_prod_host is correct already

import axios from "axios";

const Backend = axios.create({
  baseURL: process.env.REACT_APP_BACKEND_HOST_PROD,
  withCredentials: true,
});

export { Backend };

**Comment by amyegan** [comment, marked as answer]
Hey @aircliu. Since it looks like the `Access-Control-Allow-Origin` header is still missing, you might need to add it. I'm sure there are ways to do this with Flask. I just don't do enough with Python to be able to make a suggestion there. However, I know you can add CORS headers using `vercel.json`: https://vercel.com/guides/how-to-enable-cors#enabling-cors-using-vercel.json
